### PR TITLE
EN 962 categories list

### PR DESCRIPTION
**Pull request by mcarta-rt**

### CODE REVIEW

**Comment by entando-service** [comment]
All is well

File: sass/labels/list/LabelsAndLanguagesPage.scss [hunk 1]
```diff
@@ -2,27 +2,49 @@
 
 // Bootstrap overrides
 // sass-lint:disable force-element-nesting, no-qualifying-elements, class-name-format
-.LabelsAndLanguagesPage .nav-tabs > li.LabelsAndLanguagesPage__header-tab > a {
-  height: 57px;
-  line-height: 43px;
-}
+// .LabelsAndLanguagesPage .nav-tabs > li.LabelsAndLanguagesPage__header-tab > a {
```
Comment (alessioscalici) [inline sass/labels/list/LabelsAndLanguagesPage.scss]: Why adding comments?

File: src/locales/it.js [hunk 2]
```diff
@@ -68,8 +68,9 @@ export default {
     'menu.users.authority': 'Autorizzazioni',
     'menu.details': 'Dettagli',
     'menu.groups': 'Gruppi',
-    'menu.uxPattern.addLabels': 'etichette di sistema',
     'menu.labelsAndLanguages': 'Etichette e Lingue',
+    'menu.categories': 'Categorie',
+    'menu.uxPattern.addLabels': 'etichette di sistema',
```
Comment (alessioscalici) [inline src/locales/it.js]: Shouldn't Etichette be capitalized?

File: src/state/categories/reducer.js [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+      };
+    }
+    default: return state;
+  }
+};
+
+export default combineReducers({
+  list,
+  map: categoryMap,
+  // loading,
```
Comment (alessioscalici) [inline src/state/categories/reducer.js]: Please remove this comment if not necessary

File: src/state/categories/types.js [hunk 4]
```diff
@@ -0,0 +1,6 @@
+export const SET_CATEGORIES = 'categories/set-categories';
+export const SET_CATEGORY_SETTING = 'categories/set-category-setting';
+export const TOGGLE_CATEGORY_EXPANDED = 'categories/toggle-category-expanded';
+export const SET_CATEGORY_LOADING = 'categories/set-category-loading';
+export const SET_CATEGORY_LOADED = 'categories/set-category-loaded';
+export default SET_CATEGORIES;
```
Comment (alessioscalici) [inline src/state/categories/types.js]: Why SET_CATEGORIES is also exported as default? Please remove default export, we should only use named exports for categories

File: src/ui/categories/common/CategoryExpandedIcon.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
```
Comment (alessioscalici) [inline src/ui/categories/common/CategoryExpandedIcon.js]: This is basically a copy of PageExpandedIcon, why not moving PageExpandedIcon to common directory, renaming it in "TreeNodeExpandedIcon" and reuse it in both places?

File: src/ui/categories/common/CategoryFolderIcon.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
```
Comment (alessioscalici) [inline src/ui/categories/common/CategoryFolderIcon.js]: This is basically a copy of PageFolderIcon, why not moving PageFolderIcon to common directory, renaming it in "TreeNodeFolderIcon" and reuse it in both places?

File: src/ui/categories/list/CategoryListMenuActions.js [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+        </MenuItem>
+        <MenuItem
+          className="CategoryListMenuAction__menu-item-edit"
+          onClick={this.handleClick()}
+        >
+          <FormattedMessage id="app.reload" />
+        </MenuItem>
+        <MenuItem
+          className="CategoryListMenuAction__menu-item-delete"
+          onClick={this.handleClick(onClickDelete)}
```
Comment (alessioscalici) [inline src/ui/categories/list/CategoryListMenuActions.js]: ev.preventDefault is useless when using MenuItem. Please remove handleClick (you can directly pass handlers)

File: src/ui/categories/list/CategoryTree.js [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+        }
+      };
+      const className = ['CategoryTree__tree-column-td'];
+      if (category.isEmpty) {
+        className.push('CategoryTree__tree-column-td--empty');
+      }
+
+      return (
+        <tr key={category.code} className="CategoryTree__row">
+          <td className={className.join(' ').trim()}>
```
Comment (alessioscalici) [inline src/ui/categories/list/CategoryTree.js]: trim() is not necessary here (join(' ') only places spaces between array elements, it does not add leading or trailing spaces), please remove it

File: src/ui/categories/list/CategoryTree.js [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+            >
+              <CategoryExpandedIcon expanded={category.expanded} />
+              <CategoryFolderIcon empty={category.isEmpty} />
+              <span className="CategoryTree__category-name">
+                { category.title }
+              </span>
+              <RowSpinner loading={!!category.loading} />
+            </span>
+          </td>
+          {/* to move into CategoryListMenuActions */}
```
Comment (alessioscalici) [inline src/ui/categories/list/CategoryTree.js]: I guess this comment should be removed, because CategoryListMenuActions is into place

**Comment by entando-service** [comment]
All is well

**Comment by entando-service** [comment]
All is well

**Pull request merged**